### PR TITLE
Check resource file is readable

### PR DESCRIPTION
**Pull request by dhood**

### CODE REVIEW

**Comment by wjwwood** [comment]
lgtm

File: ament_index_python/ament_index_python/__init__.py [hunk 1]
```diff
@@ -43,23 +43,29 @@ def get_resource(resource_type, resource_name):
...
     for path in get_search_paths():
         resource_path = os.path.join(path, RESOURCE_INDEX_SUBFOLDER, resource_type, resource_name)
         if os.path.isfile(resource_path):
-            with open(resource_path, 'r') as h:
-                content = h.read()
-            return content
+            try:
+                with open(resource_path, 'r') as h:
+                    content = h.read()
+                return content
```
Comment (dirk-thomas) [inline ament_index_python/ament_index_python/__init__.py]: The return could be moved outside of the try / except block.

Comment (dhood) [inline ament_index_python/ament_index_python/__init__.py]: thanks - it's changed now

**Comment by dirk-thomas** [comment]
+1

**Comment by tfoote** [comment]
There's an os.access method that can check readability w/o actually opening and reading the file. https://docs.python.org/3/library/os.html#os.access

**Comment by dirk-thomas** [comment]
I think the more Pythonic way is to try and handle the exception which is what the patch already does.

**Pull request merged**